Src/FluentAssertions/Common/TypeExtensions.cs

```diff
@@ -136,6 +136,11 @@ public static bool IsEquivalentTo(this IMember property, IMember otherProperty)
 
         public static Type[] GetClosedGenericInterfaces(Type type, Type openGenericType)
         {
+            if (Type.GetTypeCode(type) != TypeCode.Object)
```

Comment:
Regrettably, this one location got missed in that PR. This change is more generic than the one in that PR. Should I undo the original change to `GenericDictionaryEquivalenceStep.cs`, so that the work is being done only in the more general place?

Reply:
I have made a commit that corrects a contractual accuracy problem with the optimization, and eliminates its duplication from the less-general location.

fd132e5fc3996028b83a3c461df9405b5d3de86fq

Reply:
I have made a commit that corrects a contractual accuracy problem with the optimization, and eliminates its duplication from the less-general location.

fd132e5fc3996028b83a3c461df9405b5d3de86f